pysingfel/detector/autoranging.py

```diff
@@ -26,36 +26,38 @@ def setupMatrices(self):
         self.offsets = self.matricize(self.offsets)
         self.switchPoints = self.matricize(self.switchPoints, None, self.switchPointVariations)
 
-    def matricize(self, array, relativeSmear=None, absoluteSmear=None): # better to separate out smearing function
+    def matricize(self, array, relativeSmear=None, absoluteSmear=None):
```

Comment:
a function should do one thing. here it does two things, that's why we want to separate this out.
def matricize(self, array)
and
def smear(array or matrix, relative, absolute)

Reply:
def matricize(self, array)
and 
def smear(self, matrices, relativeSmear=None, absoluteSmear=None)